addons/power/functions/fnc_initBattery.sqf

```diff
@@ -18,48 +18,50 @@ private _entity = _battery;
 
 if(_internal) then
 {
-	_entity = _battery getVariable 'AE3_power_parent';
+	_entity = _battery getVariable "AE3_power_parent";
 };
 
-if(!isDedicated) then {
+if(!isDedicated) then
+{
 
 	private _check = ["AE3_PowerAction", localize "STR_AE3_Power_Interaction_CheckBatteryCharge", "", 
 				{
-					params ['_target', '_player', '_params']; 
-					_params params ['_battery'];
+					params ["_target", "_player", "_params"]; 
+					_params params ["_battery"];
 					_handle = [_battery] spawn AE3_power_fnc_checkBatteryLevelAction;
 				}, 
 				{alive _target},
 				{},
 				[_battery]
 				] call ace_interact_menu_fnc_createAction;
 
-	[_entity, 0, ["ACE_MainActions", "AE3_DeviceAction"], _check] call ace_interact_menu_fnc_addActionToObject;
+	private _parentActionPath = _entity getVariable ["AE3_parentActionPath", ""];
```

Comment:
Why don't u use addParentAction instead?

Reply:
I don't have access to the `_name` variable. To achieve that I would need to pass additional parameters to this function. Because of the way the code flows, this variable is already set and can easily be used.